miden/tests/integration/operations/u32_ops/arithmetic_ops.rs

```diff
@@ -145,40 +146,32 @@ fn u32addc_fail() {
     let test = build_op_test!(asm_op, &[0, U32_BOUND, 0]);
     test.expect_error(TestError::ExecutionError("NotU32Value"));
 
-    // should fail if c > 1
-    let test = build_op_test!(asm_op, &[2, 0, 0]);
-    test.expect_error(TestError::ExecutionError("NotBinaryValue"));
+    // should fail if c >= 2^32
+    let test = build_op_test!(asm_op, &[U32_BOUND, 0, 0]);
+    test.expect_error(TestError::ExecutionError("NotU32Value"));
 }
 
 #[test]
-fn u32addc_unsafe() {
-    let asm_op = "u32addc.unsafe";
+fn u32add3_unsafe() {
+    let asm_op = "u32add3.unsafe";
 
     // --- test correct execution -----------------------------------------------------------------
-    // should push d = (a + b + c) % 2^32 onto the stack, where c is 1 or 0
-    // should push overflow flag e, where e = 1 if (a + b + c) >= 2^32 and e = 0 otherwise
-    test_addc(asm_op);
+    // should push d = (a + b + c) % 2^32 onto the stack.
+    // should push overflow flag e, where e = 2 if (a + b + c) >= 2^64, e = 1 if (a + b + c)
+    //  >= 2^32 and e = 0 otherwise.
```

Comment:
This needs to be updated since `(a + b + c) >= 2^64` can never happen.

Reply:
Agreed. It should be `2^33` but floor sounds better. I've corrected it in the latest PR.

Please see [c01f9c9](https://github.com/maticnetwork/miden/pull/221/commits/c01f9c9f740150f19ac3b58150701c24a1941733).